frmts/hdf5/s102dataset.cpp

```diff
@@ -412,34 +424,45 @@ GDALDataset *S102Dataset::Open(GDALOpenInfo *poOpenInfo)
 }
 
 /************************************************************************/
-/*                       OpenQualityOfSurvey()                          */
+/*                       OpenQuality()                          */
 /************************************************************************/
 
-bool S102Dataset::OpenQualityOfSurvey(
-    GDALOpenInfo *poOpenInfo, const std::shared_ptr<GDALGroup> &poRootGroup)
+bool S102Dataset::OpenQuality(GDALOpenInfo *poOpenInfo,
+                              const std::shared_ptr<GDALGroup> &poRootGroup)
 {
     const bool bNorthUp = CPLTestBool(
         CSLFetchNameValueDef(poOpenInfo->papszOpenOptions, "NORTH_UP", "YES"));
 
-    auto poGroupQualityOfSurvey = poRootGroup->OpenGroup("QualityOfSurvey");
-    if (!poGroupQualityOfSurvey)
+    const char *pszNameOfQualityGroup = "QualityOfSurvey";
+    auto poGroupQuality = poRootGroup->OpenGroup(pszNameOfQualityGroup);
+    if (!poGroupQuality)
     {
-        CPLError(CE_Failure, CPLE_AppDefined,
-                 "Cannot find group /QualityOfSurvey");
-        return false;
+        pszNameOfQualityGroup = "QualityOfBathymetryCoverage";
+        poGroupQuality = poRootGroup->OpenGroup(pszNameOfQualityGroup);
+        if (!poGroupQuality)
+        {
+            CPLError(CE_Failure, CPLE_AppDefined,
```

Comment:
Can we specify in the error message which group exactly was expected?

Reply:
We could based on the IHO spec version, but that's more code to add to fetch it, and I don't think it is necessary. The current message should be an already good enough hint, and it would only happen if users try to open a subdataset whose name hasn't been reported before on the main ile